service/src/query_processor.rs

```diff
@@ -91,7 +93,14 @@ impl QueryProcessor {
         let response = self
             .graph_node
             .subgraph_query_raw(&query.subgraph_deployment_id, query.query)
-            .await?;
+            .await
+            .map_err(|e| {
+                metrics::INDEXER_ERROR
+                    .with_label_values(&[&IndexerErrorCode::IE033.to_string()])
+                    .inc();
+
+                e
```

Comment:
It would be cool if `indexer_error` would automatically do this.

Ideally, we'd replicate what https://github.com/graphprotocol/indexer/blob/main/packages/indexer-common/src/errors.ts#L181-L183 does:

- Any time there is an internal error, we call `indexer_error` (or an `indexer_error!` macro), passing in the code and an optional cause).
- This internally constructs an `IndexerError` and also increments a metric.
- Whenever we call `indexer_error`, we should log the error as well, like it's done here: https://github.com/graphprotocol/indexer/blob/f20152a28bea60b0f2ed8113dfad2b45b9a500d4/packages/indexer-common/src/indexer-management/allocations.ts#L172-L174

Basically, an indexer error should allow the indexer to detect it via metrics or logs.

Reply:
Yeah cool idea! 

I moved metrics recording and logs into `indexerError::new` so they are automatically taken care of when an `indexerError` is created.